tablecodec/tablecodec.go

```diff
@@ -276,6 +276,19 @@ func DecodeKeyHead(key kv.Key) (tableID int64, indexID int64, isRecordKey bool,
 	return
 }
 
+// DecodeIndexID decodes indexID from the key.
+// this method simply extract index id part, and no other checking.
+// Caller should make sure the key is an index key.
+func DecodeIndexID(key kv.Key) (int64, error) {
```

Comment:
seems in below `DecodeTableID` we should also consider keyspace of APIv2

Reply:
only that method checks it, seems we don't need to check it here for index kv
```
	// `V2` use new encoding for RawKV & TxnKV to support more features.
	//
	// Key Encoding:
	//  TiDB: start with `m` or `t`, the same as `V1`.
	//  TxnKV: prefix with `x`, encoded as `MCE( x{keyspace id} + {user key} ) + timestamp`.
	//  RawKV: prefix with `r`, encoded as `MCE( r{keyspace id} + {user key} ) + timestamp`.
```

Reply:
Oh, keyspace may be automatically adjusted in client-go, we don't need to care about it here.